### DISCUSSION TITLE
How to display source code for wrapped component?

### DISCUSSION THREAD

**Discussion by runoncedev** [storybookjs/storybook#25525] [Help]
### Summary

I've a component called `Input`, and I've created a `WrappedInput` inside a stories file:

```
const WrappedInput = () => {
  const [value, setValue] = useState('Foo');

  return <Input value={value} onChange={(event) => setValue(event.currentTarget.value)} />;
}

export const ControlledValue = {
  render: () => <WrappedInput />,
}
```

But when I click on `Show code` inside the story I get:

```
{
  render: () => <WrappedInput />,
}
```

Instead I want to get:

```
const [value, setValue] = useState('Foo');

return <Input value={value} />;
```

I can pass a custom source code by using `parameters.source.code` but I was wondering if there might be a better approach.

Also tried passing `WrappedInput` to `parameters.source.code` but this yields the compiled code instead of the original source code.

I found a similar question but it was more focussed on managing the state:

* https://github.com/storybookjs/storybook/discussions/21680

There is also an addon that could help but it haven't been updated in a long time:

* https://storybook.js.org/addons/storybook-addon-state/

Any ideas?

**Comment by abaracedo** [comment]
I have the same issue, I have a component that needs to be rendered using the `render` function. In one of the stories I also added the `parameters.docs.description` property to add a description for that specific scenario.

When I click to view the code the result in the Source block is exactly the code I wrote describing the story.

![image](https://github.com/storybookjs/storybook/assets/9487675/2f506d2c-e55b-4833-b35a-96236287ed27)
